Sources/Aardvark/BugReporting/LogStoreAttachmentGenerator.swift

```diff
@@ -124,10 +129,19 @@ public final class LogStoreAttachmentGenerator: NSObject {
 
         let fileName = logsFileName(for: logStoreName, fileType: "txt")
 
+        let recentErrorSummary = logMessages
+            .reversed()
+            .lazy
+            .filter { $0.type == .error }
+            .prefix(numberOfErrorsInHighlights)
+            .map { $0.text }
+            .joined(separator: "\n")
```

Comment:
A few notes on this:

* `reversed()` is always lazy, true, but without already knowing this or referring to the docs, it seems as though you're intentionally not making that operation lazy given the following `lazy` call. I think it would be a bit clearer if you flipped the order and did `lazy.reversed().filter`. Apple uses that order in their `ReversedCollection` docs [here](https://developer.apple.com/documentation/swift/reversedcollection) fwiw
* Do we definitely want the highlighted logs in reverse chronological order? You could potentially drop the `reversed()` call entirely and then use `suffix()` instead of `prefix()` (i.e. `lazy.filter.suffix.map.joined`). That would be a breaking change, but we're already in breaking change mode after all
* If we decide to keep reverse chronological order, it would potentially make things clearer to remove the initial `reversed()` call and add it back in after `suffix`, i.e. `lazy.filter.suffix.reversed.map.joined`. The clarity of that is more subjective though

Reply:
> I think it would be a bit clearer if you flipped the order and did `lazy.reversed().filter`

Fair enough, that's an easy change to increase readability 👍 

> Do we definitely want the highlighted logs in reverse chronological order?

I think the theory here is that the most recent error is probably the one that's relevant to the report, so should be the first one you want to see highlighted. In other words, the errors in the highlights summary aren't intended to represent a sequence of chronological events exactly, but rather individual messages surfaced in order of (presumed) relevance.

> it would potentially make things clearer to remove the initial `reversed()` call and add it back in after `suffix`

I don't have a strong preference for either ordering. Happy to change it if you think that's easier to read.